pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,262 @@
+
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/image2.jpg', 500, 604, layout='intrinsic', alt='Image of basic amp script tutorial starter app with password requirements', align='' ) }}
+
+Play around with different passwords! If you press the submit button before all requirements are met `<amp-form>`’s [`patten`](https://amp.dev/documentation/components/amp-form?format=websites#verification) attribute will throw an error.
+
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/image3.jpg', 500, 605, layout='intrinsic', alt='Image of basic amp script tutorial starter app with unhelpful error', align='' ) }}
+
+Frustratingly, it will not explain what is missing. The tool tip only states the required pattern was not met. With the introduction of `<amp-script>` we can build out additional functionality that evaluate the user’s input and communicate what is missing! 
+
+
+# Import `amp-script`
+
+Like nearly all AMP components, `<amp-script>` requires a script tag. Open `index.html` and add the `amp-script` script tag to the head of the document.
+
+
+```html
+<head>
+ ... 
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+  ...
+</head>
+
+```
+
+
+The `amp-script` tags must be wrapped around the elements it wishes to mutate and interact with. `amp-script` is unable to mutate or interact with anything that is not a direct child of it.
+
+Our functionality relays and changes the DOM of our `form`. Place the `<form>` element inside the `<amp-script>` component. 
+
+
+```html
+<amp-script src="http://localhost:8080/js/script.js">
+    <form method="post"
+    action-xhr="/form"
+    target="_top"
+    class="card">
+        . . . 
+    </form>
+</amp-script>
+```
+
+[tip type="note"]
+**Note**: Currently, the `src` attribute must point to an absolute URL.
+[/tip]
+
+The `src` attribute points to the filepath `http://localhost:8080/js/script.js`. Create a directory titled `js` and add the `script.js` file. 
+
+
+# Inject custom script
+
+Open the newly created `script.js` file and add `console.log("amp-script here")`. Reload the page and open the [DevTools console](https://developers.google.com/web/tools/chrome-devtools/) to verify it successfully logged "amp-script here". 
+
+[tip type="important"]
+**Important**: amp-script is still in experimental mode. You will need to enable it inside the console by running `AMP.toggleExperiment('amp-script')` and confirming that it returns `true`. Read more about experimental components [here](https://amp.dev/documentation/guides-and-tutorials/learn/experimental?referrer=ampproject.org). 
```

Comment:
Better to remove `referrer=ampproject.org` from URL, as we use this to track who's still coming from ampproject.org